cdap-authorization-dataset-plugin/src/main/java/co/cask/cdap/security/authorization/ACLDataset.java

```diff
@@ -115,6 +116,28 @@ public void remove(EntityId entity) {
     }
   }
 
+  public Set<Privilege> listPrivileges(Principal principal) {
+    Set<Privilege> privileges = new HashSet<>();
+    // scan the whole table
+    Scanner scan = table.scan(null, null);
```

Comment:
I think since the principal is in middle of the key this should be the only way with the current table implementation. 
I will also ask @poornachandra if I am unaware of something.